config.template.js

```diff
@@ -4,68 +4,59 @@ const G3W_HOST_SCHEMA  = 'http';
-  distFolder:    './dist',        // path to G3W-CLIENT dist folder
-  clientFolder:  './dist/client', // path to G3W-CLIENT client folder
-  admin_plugins_folder:    G3W_ADMIN_PATH, // path to G3W-ADMIN where are stored all plugin folders
-  admin_static_folder:    `${G3W_ADMIN_PATH}/client/static`,    // path to G3W-ADMIN client/static
-  admin_templates_folder: `${G3W_ADMIN_PATH}/client/templates`, // path to G3W-ADMIN client/templates
+  assetsFolder:  './src/assets',                                            // path to G3W-CLIENT assets folder
+  pluginsFolder: './src/plugins',                                           // path to G3W-CLIENT plugins folder
+  admin_plugins_folder:   '../g3w-admin/g3w-admin',                         // path to G3W-ADMIN plugins folder
+  admin_overrides_folder: '../g3w-suite-docker/config/g3w-suite/overrides', // path to G3W-SUITE overrides folder
   host: G3W_HOST,
   port: G3W_CLIENT_PORT,
   // proxy configuration for local G3W_ADMIN server (where G3W-ADMIN is running)
   proxy: {
     host: G3W_HOST,
     url: `${G3W_HOST_SCHEMA}://${G3W_HOST}:${G3W_ADMIN_PORT}/`,
-    routes: G3W_PROXY_ROUTES
   },
   test: {
     path: '/test/config/groups/'
   },
-  createProject: {
-    before(project) { /* code here */ },
-    after(project) { /* code here */ },
-  },
-  setCurrentProject: {
-    before(project) { /* code here */ },
-    after(project) { /* code here */ },
-  },
-  // override "initConfig->group->plugins" attribute for custom plugin development
   plugins: G3W_PLUGINS,
-  keys: G3W_KEYS
+  keys: G3W_KEYS,
+  devConfig: function() {
+    g3wsdk.core.ApplicationService.once('ready', () => { });
+    g3wsdk.core.project.ProjectsRegistry.onbefore('createProject', (projectConfig) => {});
+    g3wsdk.core.project.ProjectsRegistry.onafter('createProject', (projectConfig) => {});
+    g3wsdk.core.project.ProjectsRegistry.onbefore('setCurrentProject', (project) => {});
+    g3wsdk.core.project.ProjectsRegistry.onafter('setCurrentProject', (project) => {});
+    g3wsdk.gui.GUI.once('ready', () => { console.log('ready'); });
+  }
 };
 
 // backward compatibilities (v3.x)
 if (version < '4') {
-  conf.assetsFolder    = (version < '3.7' ? './assets' : conf.assetsFolder);
-  conf.proxy.urls      = conf.proxy.routes;
-  conf.localServerPort = conf.port;
-  conf.g3w_admin_paths = {
+  conf.assetsFolder           = (version < '3.7' ? './assets' : conf.assetsFolder);
+  conf.distFolder             = './dist';
+  conf.clientFolder           = './dist/client';
+  conf.admin_static_folder    = `${conf.admin_plugins_folder}/client/static`;
+  conf.admin_templates_folder = `${conf.admin_plugins_folder}/client/templates`;
+  conf.createProject          = { before() {}, after() {} };
+  conf.setCurrentProject      = { before() {}, after() {} };
+  conf.plugins                = conf.plugins.reduce((a, v) => ({ ...a, [v]: { gid: 'qdjango:1', baseurl: `.${conf.distFolder}` }}), {});
```

Comment:
@Raruto this line raise an error :+1: 
`  conf.plugins                = conf.plugins.reduce((a, v) => ({ ...a, [v]: { gid: 'qdjango:1', baseurl: `.${conf.distFolder}` }}), {});
                                             ^
TypeError: conf.plugins.reduce is not a function
`